src/core_modules/capture-core/storageControllers/cache.types.js

```diff
@@ -153,6 +153,12 @@ export type CachedTrackedEntityTypeTranslation = {
     locale: string,
     value: string,
 };
+export type CachedProgramSection = {
+    id: string,
+    displayFormName: string,
+    sortOrder: number,
+    trackedEntityAttributes: Array<string>
```

Comment:
Is there any reason why this `trackedEntityAttributes` is array of string while `trackedEntityAttributes` in  `apiProgramSections ` is array of object. I see these 2 type are quite identical ?

Reply:
Good question! This is because the api returns objects of each trackedEntityAttribute, but this is already stored in the cache when the program itself is fetched. So to prevent double storage, I filter just the ids of each attribute and then match them with the ids of the attributes stored in the cache when the enrollment-form is built. This filtering is done in the `convertProgramSections` in `storePrograms.js` 👍 Let me know if anything is unclear!

Reply:
And this is the matching of the ids ☺️
<img width="804" alt="image" src="https://user-images.githubusercontent.com/12842868/120768994-fb408d80-c51c-11eb-9c72-f5ef507aa294.png">

Reply:
yep! thanks for the explanation